pandas/core/generic.py

```diff
@@ -2249,17 +2250,18 @@ def to_excel(
 
     def to_json(
         self,
-        path_or_buf=None,
-        orient=None,
-        date_format=None,
-        double_precision=10,
-        force_ascii=True,
-        date_unit="ms",
-        default_handler=None,
-        lines=False,
-        compression="infer",
-        index=True,
-    ):
+        path_or_buf: Optional[FilePathOrBuffer] = None,
+        orient: Optional[str] = None,
+        date_format: Optional[str] = None,
+        double_precision: int = 10,
+        force_ascii: bool_t = True,
+        date_unit: str = "ms",
+        default_handler: Optional[Callable[[Any], Union[Scalar, List, Dict]]] = None,
+        lines: bool_t = False,
+        compression: Optional[str] = "infer",
+        index: bool_t = True,
+        indent: Optional[int] = None,
+    ) -> Optional[str]:
```

Comment:
Nit [no action needed]: I know that we've been pushing typing, but in the future, it would be preferable to separate out this effort from other changes like this enhancement.  Makes things easier to review.

Reply:
Yea for sure - these do get kind of complicated when you annotate a new module altogether. Hopefully less of an issue going forward especially with all of the work @simonjayhawkins has been putting in